### PR TITLE
Feat(calsensor)!: Seed static_random from iCal UID

### LINKED ISSUE #409: Populate CalendarEvent.uid from iCal source UID
## Summary

The `CalendarEvent` dataclass in Home Assistant has a `uid` field (`str | None = None`) that calendar integrations can populate from the underlying iCal UID. Currently, Rental Control's `_ical_event()` method in `coordinator.py` creates `CalendarEvent` objects without setting this field, so it defaults to `None`.

## Problem

Downstream integrations (e.g., TurnoverCal) that consume Rental Control's calendar events need stable identifiers that persist across iCal calendar refreshes — even when event times change (e.g., a guest modifies their reservation). Without the original iCal UID, downstream consumers must derive identifiers from event content (summary + times), which breaks when those fields change.

## Proposed Fix

In `coordinator.py`, the `_ical_event()` method receives the parsed iCal `event` dict which should contain a `"UID"` key from the icalendar parser. Pass it through to `CalendarEvent`:

```python
cal_event = CalendarEvent(
    description=description,
    end=end.astimezone(self.timezone),
    location=event.get("LOCATION"),
    summary=event.get("SUMMARY", "Unknown"),
    start=start.astimezone(self.timezone),
    uid=str(event.get("UID", "")),  # <-- Add this
)
```

This is a non-breaking change — the field is optional and adding it simply exposes data that already exists in the parsed iCal events.

## Context

- HA's `CalendarEvent` added the `uid` field to support calendar integrations that have stable event identifiers
- The underlying icalendar parser preserves the `UID` property from RFC 5545 VEVENT components
- The TurnoverCal integration needs this to satisfy its UID Stability Contract (turnover event UIDs must remain stable across source event time changes)

### PR DESCRIPTION
**Pull request by tykeal**
## Summary

Switch the `static_random` door code generator to seed from the iCal UID
instead of the event description. This stabilizes generated codes when
the calendar description is edited without changing the underlying event
identity.

Falls back to description seeding when UID is unavailable.

## Breaking Change

⚠️ **static_random codes will rotate once** for all events on the first
refresh after this update. After that initial rotation, codes remain
stable across description edits as long as the event UID doesn't change.

### Commit 1: Expose iCal UID as event attribute
- Added `uid` to `_event_attributes` initialization
- Populated from `event.uid` during coordinator updates
- Reset on no-events path

### Commit 2: Seed static_random from UID (BREAKING)
- Changed `_generate_door_code()` to seed from UID with description
  fallback
- Updated description-is-None guard: when generator is `static_random`
  and UID is available, don't force fallback to `date_based`
- Added 7 new tests covering deterministic seeding, stability across
  description changes, UID-differs codes, fallback to description,
  and fallback to date_based

### Commit 3: Spec artifacts
- Added `specs/001-static-random-uid-seed/` with full speckit artifacts

## Testing

All 632 tests pass. 7 new tests added for static_random UID seeding.

Closes #409

### CODE REVIEW

File: custom_components/rental_control/sensors/calsensor.py [hunk 1]
```diff
@@ -373,6 +385,7 @@ def _handle_coordinator_update(self) -> None:
             self._event_attributes["end"] = event.end
             self._event_attributes["location"] = event.location
             self._event_attributes["description"] = event.description
+            self._event_attributes["uid"] = event.uid if hasattr(event, "uid") else None
```
[the remark this answers is not in the document]
Comment (tykeal) [reply custom_components/rental_control/sensors/calsensor.py]: Addressed in commit 2e29222: normalized empty-string UIDs to None at capture time and switched to explicit fallback chain in `_generate_door_code()`.

File: custom_components/rental_control/sensors/calsensor.py [hunk 2]
```diff
@@ -276,8 +284,12 @@ def _generate_door_code(self) -> str:
...
-            # If the description changes this will most likely change the code
-            random.seed(self._event_attributes["description"])
+            # Prefer UID (immutable per RFC 5545) over description (mutable)
+            seed = (
+                self._event_attributes.get("uid")
+                or self._event_attributes["description"]
+            )
+            random.seed(seed)
             max_range = int("9999".rjust(code_length, "9"))
             ret = str(random.randrange(1, max_range, code_length)).zfill(code_length)
```
[the remark this answers is not in the document]
Comment (tykeal) [reply custom_components/rental_control/sensors/calsensor.py]: Addressed in commit 2e29222: explicit fallback chain now ensures `random.seed()` is never called with None. Also added a test for the empty-string UID edge case.
